### PR TITLE
GAWB-947: billing project dropdown should only display project names

### PR DESCRIPTION
**Pull request by MatthewBemis**
- [x] **Submitter**: Rebase to develop. DO NOT SQUASH
- [x] **Submitter**: Make sure Swagger is updated if API changes
- [x] **Submitter**: Make sure documentation for code is complete
- [x] **Submitter**: Review code comments; remove done TODOs, create stories for remaining TODOs
- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Add description or comments on the PR explaining the hows/whys (if not obvious)
- [x] **Submitter**: Tell tech lead that the PR exists if s/he wants to look at it
- [x] **Submitter**: Anoint a lead reviewer (LR). **Assign PR to LR**
- [ ] **LR**: Initial review by LR and others.
- [ ] Comment / review / update cycle:
  - Rest of team may comments on PR at will
  - **LR assigns to submitter** for feedback fixes
  - Submitter updates documentation as needed
  - Submitter rebases to develop again if necessary
  - Submitter makes further commits. DO NOT SQUASH. **Reassign to LR** for further feedback
- [x] **Tech lead**: sign off
- [ ] **LR**: sign off
- [ ] **Assign to submitter** to finalize
- [ ] **Submitter**: Squash commits, rebase if necessary
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Merge to develop 
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: Check configuration files in Jenkins in case they need changes
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't 
  working for ANY reason!
- [ ] **Submitter**: Verify swagger UI on dev server still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via hipchat and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: src/cljs/org/broadinstitute/firecloud_ui/page/workspace/create.cljs [hunk 1]
```diff
@@ -106,7 +106,10 @@
       {:endpoint (endpoints/get-billing-projects)
        :on-done (fn [{:keys [success? get-parsed-response]}]
                   (if success?
-                    (let [billing-projects (get-parsed-response)]
+                    (let [billing-projects (flatten (->> (get-parsed-response)
+                                                         (map utils/keywordize-keys)
+                                                         (map #(select-keys % [:projectName]))
+                                                         (map #(vals %))))]
```
Comment (MattPutnam) [inline src/cljs/org/broadinstitute/firecloud_ui/page/workspace/create.cljs]: This is way overkill.  All you need is

``` clojure
(let [billing-projects (map #(% "projectName") (get-parsed-response))]
  ...)
```

**Comment by MattPutnam** [comment]
Hmm... I noticed that the get-billling-projects endpoint is being called twice.  Looks like WorkspacesList calls it and passes it to WorkspaceTable, but WorkspaceTable doesn't pass it to create/Button and create/Button hits it again.  Can you update this so that create/Button takes the projects as a prop and makes no ajax call on load?

**Comment by MatthewBemis** [comment]
@MattPutnam back to you

File: src/cljs/org/broadinstitute/firecloud_ui/page/workspace/create.cljs [hunk 2]
```diff
@@ -98,17 +98,11 @@
...
-                      (swap! state assoc :billing-projects billing-projects)
-                      (if (empty? billing-projects)
-                        (swap! state assoc :disabled-reason :no-billing)
-                        (swap! state dissoc :disabled-reason)))
-                    (swap! state assoc :disabled-reason :error)))}))})
+   (fn [{:keys [props state]}]
+     (let [disabled-reason (:disabled-reason props)]
+       (if (some? disabled-reason)
+         (swap! state assoc :disabled-reason disabled-reason)
+         (swap! state dissoc :disabled-reason))))})
```
Comment (MattPutnam) [inline src/cljs/org/broadinstitute/firecloud_ui/page/workspace/create.cljs]: This chunk doesn't seem quite right.  Now that all of this is passed in there shouldn't be a need for `:component-did-mount` at all.  This check should be done where the button is rendered.

File: src/cljs/org/broadinstitute/firecloud_ui/page/workspaces_list.cljs [hunk 3]
```diff
@@ -194,7 +196,7 @@
          :else
          [:div {:style {:margin "0 2em"}}
           [WorkspaceTable
-           (assoc props :workspaces workspaces :billing-projects billing-projects)]])))
+           (assoc props :workspaces workspaces :billing-projects billing-projects :disabled-reason error-message)]])))
```
Comment (MattPutnam) [inline src/cljs/org/broadinstitute/firecloud_ui/page/workspaces_list.cljs]: I don't think this is right.  The error message is kind of an infodump; this flag was designed to be a keyword (`:no-billing` etc.)

**Comment by MatthewBemis** [comment]
@MattPutnam back to you with latest fixes

**Comment by MattPutnam** [comment]
@MatthewBemis Check out what I just pushed and let me know if you have any questions.  WorkspaceList is using David Mohs's pattern of putting all ajax responses in a `:server-response` within state, so the disabled reason wasn't hooked up right.  Also, you weren't detecting the empty case.  Then I moved the map so that we're never holding onto more info than we need.

**Comment by MatthewBemis** [comment]
@MattPutnam nice catch with the empty case- totally slipped my mind to re-implement that bit. the server-response pattern was a bit unclear to me at first but i see with your commit how to work with it now. thanks for the help!

**Comment by MattPutnam** [comment]
👍

**Comment by davidangb** [comment]
👍 as TL

**Pull request merged**